public/app/features/dashboard-scene/serialization/transformSaveModelV2ToV1.test.ts

```diff
@@ -0,0 +1,315 @@
+      // Remove metadata fields that may differ between backend and frontend transformations
+      const frontendSpec = { ...frontendOutputAfterLoadedByScene };
+      const backendSpec = { ...backendOutputAfterLoadedByScene };
+
+      // Remove metadata fields that are not part of the core dashboard spec
+      delete frontendSpec.uid;
+      delete backendSpec.uid;
+      delete frontendSpec.version;
+      delete backendSpec.version;
+      delete frontendSpec.id;
+      delete backendSpec.id;
+
+      // Compare only the spec structures - this is the core transformation
+      expect(backendSpec).toEqual(frontendSpec);
+    });
+  });
+});
+
+/*
+ * Simulate the frontend transformation of a dashboard data object to a v1beta1 dashboard data object
+ * This is to ensure that the frontend transformation produces the same result as the backend transformation
+ * when the dashboard data object is loaded by the scene.
+ */
+/*
+ * Loads a v1beta1 dashboard into Scene and serializes it back to v1beta1.
+ *
+ * This simulates the real-world flow when editing a dashboard:
+ * v1beta1 JSON → DashboardModel → Scene → v1beta1 JSON
+ *
+ * This function is used to normalize both backend and frontend outputs
+ * through the same Scene load/save cycle, ensuring we compare apples to apples.
+ */
+function loadAndSerializeV1SaveModel(dashboard: Dashboard): Dashboard {
+  const sceneBackend = transformSaveModelToScene({
+    dashboard: dashboard as DashboardDataDTO,
+    meta: {
+      isNew: false,
+      isFolder: false,
+      canSave: true,
+      canEdit: true,
+      canDelete: false,
+      canShare: false,
+      canStar: false,
+      canAdmin: false,
+      isSnapshot: false,
+      provisioned: false,
+      version: 1,
+    },
+  });
+  const backendOutputAfterLoadedByScene = transformSceneToSaveModel(sceneBackend, false);
+
+  return backendOutputAfterLoadedByScene;
+}
+
+/*
+ * FRONTEND PATH:
+ * Transforms v2beta1 to v1beta1 using the frontend conversion pipeline:
+ * v2beta1 input → Scene (via transformSaveModelSchemaV2ToScene) → v1beta1 (via transformSceneToSaveModel)
+ *
+ * Then passes through loadAndSerializeV1SaveModel to normalize with the same Scene load/save
+ * cycle that the backend output goes through. This ensures both paths are compared after
+ * the same normalization process.
+ */
+function transformV2ToV1UsingFrontendTransformers(jsonInput: DashboardWithAccessInfo<DashboardV2Spec>): Dashboard {
+  // Step 1: Load v2beta1 into Scene
+  const scene = transformSaveModelSchemaV2ToScene({
+    spec: jsonInput.spec,
+    metadata: jsonInput.metadata || {
+      name: 'test-dashboard',
+      generation: 1,
+      resourceVersion: '1',
+      creationTimestamp: new Date().toISOString(),
+    },
+    apiVersion: jsonInput.apiVersion,
+    access: {},
+    kind: 'DashboardWithAccessInfo',
+  });
+
+  // Step 2: Transform Scene to v1beta1
+  const frontendOutput = transformSceneToSaveModel(scene, false);
+
```

Comment:
Would it be possible to use this instead of the responseTransformers.ensureV1Response (or make ensureV1Response do this instead)? Would be nice to only have one path in the frontend that converts v2 to v1